src/Scripts/DefaultVirtualScrips.spec.ts

```diff
@@ -0,0 +1,253 @@
+import 'reflect-metadata';
+import { DefaultVirtualScripts } from './DefaultVirtualScripts';
+import { VirtualScript, VirtualScriptType } from './VirtualScript';
+import { expect } from 'chai';
+
+const isValidIterable = (
+  instance: unknown
+): instance is IterableIterator<unknown> => {
+  if (instance === null || instance === undefined) {
+    return false;
+  }
+
+  return typeof instance[Symbol.iterator] === 'function';
+};
+
+describe('DefaultVirtualScripts', () => {
+  let virtualScripts: DefaultVirtualScripts;
+
+  beforeEach(() => {
+    virtualScripts = new DefaultVirtualScripts();
+  });
+
+  it('should be iterable', () => {
+    expect(isValidIterable(virtualScripts)).to.equal(true);
+  });
+
+  describe('clear', () => {
+    it('should remove all scripts when type is not passed', () => {
+      // arrange
+      virtualScripts.set('first', VirtualScriptType.LOCAL, 'let a = 1;');
+      virtualScripts.set('second', VirtualScriptType.REMOTE, 'let a = 2;');
+      // act
+      virtualScripts.clear();
+      // assert
+      expect(virtualScripts).to.deep.equal(new DefaultVirtualScripts());
+    });
+
+    it('should remove only VirtualScriptType.LOCAL scripts when VirtualScriptType.LOCAL type is passed', () => {
+      // arrange
+      virtualScripts.set('first', VirtualScriptType.LOCAL, 'let a = 1;');
+      virtualScripts.set('second', VirtualScriptType.REMOTE, 'let a = 2;');
+      const expectedVirtualScripts = new DefaultVirtualScripts().set(
```

Comment:
```suggestion
      const expected = new DefaultVirtualScripts().set(
```
And see the comment above